### DISCUSSION TITLE
Remove default settings in the GUI from Oracle/TPC-C to no database and workload

### DISCUSSION THREAD

**Discussion by jamiere-msft** [TPC-Council/HammerDB#55] [Q&A] [answered]
This is from our SQL Evangelist who does many live demos with HammerDB.  He asked if we could set the defaults in the GUI to no database and no workload.   It is a perception issue, not anything funcitonal.

**Comment by abondvt89** [comment]
Also an issue in the CLI.

**Comment by sm-shaw** [comment]
Currently this is configurable by updating the generic.xml file in the config directory.  It needs to be set to a particular database and workload to initialize HammerDB as it associates the chosen database and workload with the functions that are called, so it cannot be blank, if it is it will use the default.  Oracle/TPC-C is in the default generic.xml file, however this is only for historical reasons as this workload was done first.  To change it after install you can change generic.xml - For example this would make SQL Server and TPC-H the default on subsequent startup: 
```
   <benchmark>
        <rdbms>MSSQLServer</rdbms>
        <bm>TPC-H</bm>
   </benchmark>
```
So the question is - is this sufficient or would it be preferred to be configurable during install somehow?

**Comment by jamiere-msft** [comment, marked as answer]
Modifying the generic.xml file should be sufficient.  I will check with the person who reported it to me and confirm.

Jamie Reding | Microsoft SQL Server | 425-703-0510

**Comment by jamiere-msft** [comment]
I confirmed that this is acceptable.  Thanks!

**Comment by sm-shaw** [comment]
Issue closed as modifying generic.xml gives the desired functionality
